app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -138,6 +154,224 @@ public static void playOnBackgroundPlayer(final Context context, final PlayQueue
+                    secondaryStreamUrl = secondaryStream.getStream().getUrl();
+
+                    if (videoStream.getFormat() == MediaFormat.MPEG_4)
+                        psName = Postprocessing.ALGORITHM_MP4_FROM_DASH_MUXER;
+                    else
+                        psName = Postprocessing.ALGORITHM_WEBM_MUXER;
+
+                    psArgs = null;
+                    long videoSize = wrappedVideoStreams.getSizeInBytes(videoStream);
+
+                    // set nearLength, only, if both sizes are fetched or known. This probably
+                    // does not work on slow networks but is later updated in the downloader
+                    if (secondaryStream.getSizeInBytes() > 0 && videoSize > 0) {
+                        nearLength = secondaryStream.getSizeInBytes() + videoSize;
+                    }
+                }
+                fileName += videoStream.getFormat().getSuffix();
+                mime = videoStream.getFormat().getMimeType();
+                selectedStream = videoStream;
+                break;
+            case 's':
+                SubtitlesStream subtitlesStream = null;
+                for (SubtitlesStream currentStream : subtitleStreamsAdapter.getAll()) {
+                    if (currentStream.getLocale().equals(downloadSetting.getSubtitleLocale())) {
+                        subtitlesStream = currentStream;
+                        break;
+                    }
+                }
+                if (subtitlesStream == null) {
+                    subtitlesStream = subtitleStreamsAdapter.getItem(0);
+                }
+                if (subtitlesStream.getFormat() == MediaFormat.TTML) {
+                    psName = Postprocessing.ALGORITHM_TTML_CONVERTER;
+                    psArgs = new String[]{
+                            subtitlesStream.getFormat().getSuffix(),
+                            "false",// ignore empty frames
+                            "false",// detect youtube duplicate lines
+                    };
+                }
+                fileName += subtitlesStream.getFormat() == MediaFormat.TTML ? MediaFormat.SRT.suffix :
+                        subtitlesStream.getFormat().suffix;
+                mime = subtitlesStream.getFormat().getMimeType();
+                selectedStream = subtitlesStream;
+                break;
+            default:
+                return null;
+        }
+
+        if (secondaryStreamUrl == null) {
+            urls = new String[]{selectedStream.getUrl()};
+        } else {
+            urls = new String[]{selectedStream.getUrl(), secondaryStreamUrl};
+        }
+        StoredFileHelper storedFileHelper = downloadSetting.getStoredFileHelper();
+        try {
+            StoredDirectoryHelper storedDirectoryHelper = new StoredDirectoryHelper(activity.getContext(),
+                    storedFileHelper.getParentUri(), storedFileHelper.getTag());
+            StoredFileHelper storedFileHelperNew = storedDirectoryHelper.createFile(fileName, mime);
```

Comment:
the createFile() method make a new StoredFileHelper instance, this not create a file in the directory.
you should check if the file exists before continue:
```java
if (!storedFileHelperNew.existsAsFile()) {
    if (!storedFileHelperNew.create()) {
         // throw new IOException("cannot create the file");
    }
}
```